extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts

```diff
@@ -0,0 +1,610 @@
+		});
+
+		this.toDispose.push(this.dialog.onClosed(_ => this.initDialogPromise.resolve()));
+	}
+
+	public async openDialog(): Promise<void> {
+		let connection = await azdata.connection.getCurrentConnection();
+		if (connection) {
+			this.connectionId = connection.connectionId;
+		}
+
+		this.initializeDialog();
+
+		this.dialog.okButton.label = constants.updateProjectDialogOkButtonText;
+		this.dialog.okButton.enabled = false;
+		this.toDispose.push(this.dialog.okButton.onClick(async () => await this.handleUpdateButtonClick()));
+
+		this.dialog.cancelButton.label = constants.cancelButtonText;
+
+		getAzdataApi()!.window.openDialog(this.dialog);
+		await this.initDialogPromise;
+
+		this.tryEnableUpdateButton();
+	}
+
+	private dispose(): void {
+		this.toDispose.forEach(disposable => disposable.dispose());
+	}
+
+	private initializeDialog(): void {
+		this.initializeUpdateProjectFromDatabaseTab();
+		this.dialog.content = [this.updateProjectFromDatabaseTab];
+	}
+
+	private initializeUpdateProjectFromDatabaseTab(): void {
+		this.updateProjectFromDatabaseTab.registerContent(async view => {
+
+			const connectionRow = this.createServerRow(view);
+			const databaseRow = this.createDatabaseRow(view);
+			this.populateServerDropdown();
+
+			const sourceDatabaseFormSection = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
+			sourceDatabaseFormSection.addItems([connectionRow, databaseRow]);
+
+			const projectLocationRow = this.createProjectLocationRow(view);
+			const folderStructureRow = this.createFolderStructureRow(view);
+			const targetProjectFormSection = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
+			targetProjectFormSection.addItems([projectLocationRow, folderStructureRow]);
+
+			const actionRow = this.createActionRow(view);
+			const actionFormSection = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
+			actionFormSection.addItems([actionRow]);
+
+			this.formBuilder = <azdata.FormBuilder>view.modelBuilder.formContainer()
+				.withFormItems([
+					{
+						title: constants.sourceDatabase,
+						components: [
+							{
+								component: sourceDatabaseFormSection,
+							}
+						]
+					},
+					{
+						title: constants.targetProject,
+						components: [
+							{
+								component: targetProjectFormSection,
+							}
+						]
+					},
+					{
+						title: constants.updateAction,
+						components: [
+							{
+								component: actionFormSection,
+							}
+						]
+					}
+				], {
+					horizontal: false,
+					titleFontSize: cssStyles.titleFontSize
+				})
+				.withLayout({
+					width: '100%',
+					padding: '10px 10px 0 20px'
+				});
+
+			let formModel = this.formBuilder.component();
+			await view.initializeModel(formModel);
+			this.connectionButton?.focus();
+			this.initDialogPromise.resolve();
+		});
+	}
+
+	private createServerRow(view: azdata.ModelView): azdata.FlexContainer {
```

Comment:
looks like this is a lot of the same components that are built in the other dialogs. Doesn't have to be in this PR, but it would be good to have a base dialog class or something so that the different dialogs don't each have to implement creating these common UI components

Reply:
Agreed.  Within an extension is easy enough, but thinking more broadly: what are our options for sharing code between extensions?

Reply:
I vaguely remember @Charles-Gagnon has looked into this a bit in the past?

Reply:
*shrug* A small bit but we don't really have many options besides just doing an npm package (we might be able to use the same ts file in multiple extensions, but I've never seen that done so not sure how easy it'd be to set up). 

But I'd like to see what specific cases we have for sharing UI components like that across extensions that aren't something we would be adding to core instead before we start really digging into that. So if you have any cases let's chat and see what we can come up with.